specification.json

```diff
@@ -724,7 +724,7 @@
         {
             "id": "Requirement 5.3.3",
             "machine_id": "requirement_5_3_3",
-            "content": "`PROVIDER_READY` handlers attached after the provider is already in a ready state MUST run immediately.",
+            "content": "Client `PROVIDER_READY` handlers attached after the provider is in a ready state MUST run immediately.",
```

Comment:
Any reason to bind this to `Client`? 

Note - implementing this requirement require event handler implementation to track provider state change OR have reference to providers and check their status when handlers get registered.

Reply:
> Any reason to bind this to Client?

Yes - I added some justification for this in the non-normative section:

>_Application authors_ may attach readiness handlers to be confident that system is ready to evaluate flags.
If such handlers are attached after the provider underlying the client has already been initialized, they should run immediately.
**_API (global) handlers have no such requirement, as they are potentially bound to multiple providers and are used primarily for diagnostic purposes._**

> Note - implementing this requirement require event handler implementation to track provider state change OR have reference to providers and check their status when handlers get registered.

Yes - but just to be clear - this was also the case before this change, right?

Reply:
This is something I missed in go-sdk implementation. But it's connected to tracking providers through event handler mechanism 

And non-normative text expansion helps here to give a clear idea behind this. Hence ➕  for this

Reply:
Revisiting this again after the go-sdk implementation. I doubt whether this needs to be restricted to the client (scoped events)

Assume I only register a global provider. In my application, I add this in a startup logic. Alongside, I register an API-level handler for ready state. By the time execution reach handler registration, the provider could be in a ready state (especially if initialization is nonblocking and happens concurrently). If we enforce this spec change, then my global ready handler never gets executed. 

Besides, we want to consider adding another rule `Requirement 5.3.4 - PROVIDER_ERROR handlers attached after the provider is in a ready state MUST run immediately` 

Reason is, these events are connected to provider initialization. And initialization can yield `PROVIDER_READY` & `PROVIDER_ERROR` which IMO needs special handling.

Reply:
I agree with you on both counts. I think I will make this change.